udsoncan/connections.py

```diff
@@ -291,6 +298,74 @@ def empty_rxqueue(self):
 			self.rxqueue.get()
 
 
+class PythonCanConnection(BaseConnection, ISOTPMixin):
+
+	def __init__(self, rxid, txid, channel=None, interface=None,
+				 block_size=32, st_min=0, name=None, **kwargs):
+		BaseConnection.__init__(self, name)
+		ISOTPMixin.__init__(self, block_size, st_min)
+
+		self.rxid = rxid
+		self.txid = txid
+		self._extended = rxid > 0x7FF
+		self.bus = None
+
+		self.config = {
+			'channel': channel,
+			'bustype': interface,
+			'single_handle': True,
+			'can_filter': {
+				'can_id': rxid,
+				'can_mask': 0x1FFFFFFF if self._extended else 0x7FF,
+				'extended': self._extended
+			}
+		}
+		self.config.update(kwargs)
+
+	def open(self):
+		self.bus = can.interface.Bus(**self.config)
+		return self
+
+	def close(self):
+		self.bus.shutdown()
+		self.logger.info('Connection closed')
+		self.bus = None
+
+	def __enter__(self):
+		return self
+
+	def __exit__(self, type, value, traceback):
+		self.close()
+
+	def is_open(self):
+		return self.bus is not None
+
+	def empty_rxqueue(self):
+		while self.bus.recv(0) is not None:
+			pass
+
+	def specific_send(self, payload):
+		self.isotp_send(payload)

```

Comment:
n exception should be raised if connection is not opened. Sending without opening the conenction raise "None has no attribute sends()", which is misleading